parsing/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -3948,6 +3948,84 @@ SqlTypeNameSpec TypeName() :
     }
 }
 
+/**
+* Parse datetime types: date, time, timestamp.
+* Override the function in core parser.
+*/
+SqlTypeNameSpec DateTimeTypeName() :
+{
+    int precision = -1;
+    final SqlTypeName typeName;
+    SqlTimeZoneOption timeZoneOpt = SqlTimeZoneOption.WITHOUT_TIME_ZONE;
+}
+{
+    <DATE> {
+        typeName = SqlTypeName.DATE;
+        return new SqlBasicTypeNameSpec(typeName, getPos());
+    }
+|
+    LOOKAHEAD(2)
+    <TIME>
+    precision = PrecisionOpt()
+    timeZoneOpt = TimeZoneOption()
+    {
+        switch (timeZoneOpt) {
+           case WITHOUT_TIME_ZONE:
+              typeName = SqlTypeName.TIME;
+              break;
+           case WITH_LOCAL_TIME_ZONE:
+              typeName = SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE;
+              break;
+           case WITH_TIME_ZONE:
+              typeName = SqlTypeName.TIME_WITH_TIME_ZONE;
+              break;
+           default:
+              typeName = SqlTypeName.TIME;
+              break;
+        }
+        return new SqlBasicTypeNameSpec(typeName, precision, getPos());
+    }
+|
+    <TIMESTAMP>
+    precision = PrecisionOpt()
+    timeZoneOpt = TimeZoneOption()
+    {
+        switch (timeZoneOpt) {
+        case WITHOUT_TIME_ZONE:
+            typeName = SqlTypeName.TIMESTAMP;
+            break;
+        case WITH_LOCAL_TIME_ZONE:
+            typeName = SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE;
+            break;
+        case WITH_TIME_ZONE:
+            typeName = SqlTypeName.TIMESTAMP_WITH_TIME_ZONE;
+            break;
+        default:
+            typeName = SqlTypeName.TIMESTAMP;
+            break;
+        }
+        return new SqlBasicTypeNameSpec(typeName, precision, getPos());
+    }
+}
+
+SqlTimeZoneOption TimeZoneOption() :
```

Comment:
Why not name this TimeZoneOpt() so that it overrides the function with the same name in the core parser?

Reply:
Good point. I almost missed it. Fixed in https://github.com/googleinterns/calcite/pull/216/commits/cdaf2f8270faf269c6c56dc7f774f5492622a19b